assets/js/modules/analytics-4/components/widgets/EngagedTrafficSourceWidget.js

```diff
@@ -16,6 +16,26 @@
  * limitations under the License.
  */
 
-export default function EngagedTrafficSourceWidget() {
-	return <div>TODO: UI for EngagedTrafficSourceWidget</div>;
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+
+const { useSelect } = Data;
+
+export default function EngagedTrafficSourceWidget( { Widget, WidgetNull } ) {
```

Comment:
Please, add prop types for this widget and for all other widgets where new props are added.